app/services/settings/optimizer.test.ts

```diff
@@ -0,0 +1,337 @@
+                                params: [
+                                    {
+                                        category: CategoryName.output,
+                                        subCategory: 'Streaming',
+                                        key: 'simpleUseAdvanced',
+                                        setting: 'UseAdvanced',
+                                        lookupValueName: true,
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                ]
+            }
+        ]
+    }]);
+});
+
+test('SettingsKeyAccessor#traverseKeyDescriptions', () => {
+    setup();
+    // テスト用の最小のdescriptionを用意する
+    //  分岐があること
+    const simpleSettings: OptimizeSettings = {
+        outputMode: 'Simple',
+        videoBitrate: 12345,
+    }
+    const simpleDescriptions = filterKeyDescriptions(simpleSettings, AllKeyDescriptions);
+    expect(simpleDescriptions.length).toEqual(1);
+    expect(simpleDescriptions[0].dependents).not.toBeFalsy();
+    expect(simpleDescriptions[0].dependents.length).toEqual(2);
+
+    const advancedSettings: OptimizeSettings = {
+        outputMode: 'Advanced',
+        videoBitrate: 12345,
+    }
+    const advancedDescriptions = filterKeyDescriptions(advancedSettings, AllKeyDescriptions);
+    expect(advancedDescriptions.length).toEqual(1);
+    expect(advancedDescriptions[0].dependents).not.toBeFalsy();
+    expect(advancedDescriptions[0].dependents.length).toEqual(2);
+
+    // アクセサのmockを作る
+    let outputMode = 'Simple';
+    const accessor = {
+        findSettingValue: jest.fn().mockImplementation(
+            (settings: ISettingsSubCategory[], category: string, setting: string) => {
+                if (category === 'Untitled' && setting === 'Mode') {
+                    return outputMode;
+                }
+                // Simple value
+                if (category === 'Streaming' && setting === 'VBitrate') {
+                    return 4649;
+                }
+                // Advanced value
+                if (category === 'Streaming' && setting === 'Encoder') {
+                    return 'obs_x264';
+                }
+                if (category === 'Streaming' && setting === 'rate_control') {
+                    return 'CBR';
+                }
+                if (category === 'Streaming' && setting === 'bitrate') {
+                    return 2525;
+                }
+                return undefined;
+            }
+        ),
+        getSettingsFormData: jest.fn(),
+        findSetting: jest.fn(),
+        setSettings: jest.fn()
+    };
+    const a = new SettingsKeyAccessor(accessor);
+
+    // 分岐の選択されている側のみの値が得られること
+    outputMode = 'Simple';
+    const simpleResult = [...a.travarseKeyDescriptions(simpleDescriptions, d => [d.key, d.setting])];
+
+    expect(accessor.findSettingValue.mock.calls.length).toEqual(1);
```

Comment:
nits:
```suggestion
    expect(accessor.findSettingValue).toHaveBeenCalledTimes(1);
```
ref. https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber